azure-iot-device/tests/provisioning/internal/test_polling_machine.py

```diff
@@ -34,8 +35,10 @@
 
 
 class TestRequestResponseProvider(RequestResponseProvider):
```

Comment:
>TestRequestResponseProvider [](start = 6, length = 27)

suggest a better name.  This looks like a test class.  probably harmless, but still confusing.  
PytestWarning: cannot collect test class 'TestRequestResponseProvider' because it has a __init__ constructor